v2/tools/generator/internal/config/property_configuration.go

```diff
@@ -232,14 +232,15 @@ func (pc *PropertyConfiguration) UnmarshalYAML(value *yaml.Node) error {
 
 		// $importConfigMapMode: <string>
 		if strings.EqualFold(lastId, importConfigMapModeTag) && c.Kind == yaml.ScalarNode {
-			switch c.Value {
+			switch strings.ToLower(c.Value) {
```

Comment:
Not sold on the value of accepting any case here.

I think it's better to force us to be consistently cased and return an error than to allow anything. With that said... I like your `UpperCamelCase` more than my `lowercase`, so possibly we could drop this case-insensitivity but fix this `importConfigMapMode` to accept `Optional|Required` rather than `optional|required`? Then it's forced to be consistent with your new enum (which IMO we should also not ignore-case on?)

Reply:
I'd be happy to do a follow up PR to fix this enums case if you didnt' want to do it here because it increased PR size.

Reply:
As discussed offline, we'll leave this as-is for the moment, as most of our configuration is already case-insensitive.